### PR TITLE
New architecture

### PR DESCRIPTION
**Pull request by OlegFirst**

### CODE REVIEW

File: src/components/MainPage/MainPage.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+			[index + 2]: true,
+			[index + 3]: true
+		}));
+		
+		setInPlayCards({
+			firstIndex: currentCardIndex,
+			lastIndex: currentCardIndex - 4
+		});
+		
+		currentCardIndex -= 4;
```
Comment (captain-yossarian) [inline src/components/MainPage/MainPage.tsx]: THis is not right. You are not allowed to mutate variable which is declared out of component scope. Please keep in mind that it should be pure function

File: src/components/MainPage/MainPage.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		
+		console.log(currentCardIndex)		
+		const index = currentCardIndex;
+		
+		setRemovedCards(prev => ({
+			...prev,
+			[index]: true,
+			[index + 1]: true,
+			[index + 2]: true,
+			[index + 3]: true
```
Comment (captain-yossarian) [inline src/components/MainPage/MainPage.tsx]: What if you need to grab 10 cards instead of 4 ?

File: src/components/MainPage/MainPage.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+		}));
+		
+		setInPlayCards({
+			firstIndex: currentCardIndex,
+			lastIndex: currentCardIndex - 4
```
Comment (captain-yossarian) [inline src/components/MainPage/MainPage.tsx]: Please avoid using `magic numbers`.

File: src/components/Deck/Deck.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+
+
+
+const Deck = ({  }) => {	
+	return (
+		<section className="deck">
```
Comment (captain-yossarian) [inline src/components/Deck/Deck.tsx:6]: Please add prettier and eslint to the project

File: src/components/MainPage/MainPage.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import Cards from '../Cards/Cards';
+import InPlay from '../InPlay/InPlay';
+import Button from '../Button/Button';
+
+import { getCards } from '../../constants/preparation';
+const cardList = getCards();
+let currentCardIndex = cardList.length;
+
+const MainPage = () => {
+	const [removedCards, setRemovedCards] = useState({});
```
Comment (captain-yossarian) [inline src/components/MainPage/MainPage.tsx:11]: Please add type for `removeCards` state

**Comment by OlegFirst** [comment]
ok

**Pull request closed without merging**